src/figuresClasses.ts

```diff
@@ -1,11 +1,97 @@
-export interface Figure {}
+type Shape = `triangle` | `circle` | `rectangle`;
+type Color = `red` | `green` | `blue`;
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  public color: Color;
+
+  private a: number;
+
+  private b: number;
+
+  private c: number;
+
+  constructor(color: Color, a: number, b: number, c: number) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('invalid parameters');
```

Comment:
[CODE STYLE] - When throwing an error, provide a more detailed description in the error message to make it clear what went wrong. For instance: `throw new Error('Triangle side lengths must be greater than zero.')`

Reply:
> [CODE STYLE] - When throwing an error, provide a more detailed description in the error message to make it clear what went wrong. For instance: `throw new Error('Triangle side lengths must be greater than zero.')`

Good point, fix all error messages